src/databricks/labs/remorph/install.py

```diff
@@ -0,0 +1,191 @@
+                raise SystemExit(msg)
+
+        config = MorphConfig(
+            source=source,
+            skip_validation=skip_validation,
+            catalog_name=catalog_name,
+            schema_name=schema_name,
+            sdk_config=None,
+        )
+
+        ws_file_url = self._installation.save(config)
+        if self._prompts.confirm("Open config file in the browser and continue installing?"):
+            webbrowser.open(ws_file_url)
+        return config
+
+
+class WorkspaceInstallation:
+    def __init__(
+        self,
+        config: MorphConfig,
+        installation: Installation,
+        ws: WorkspaceClient,
+        prompts: Prompts,
+        verify_timeout: timedelta,
+    ):
+        self._config = config
+        self._installation = installation
+        self._ws = ws
+        self._prompts = prompts
+        self._verify_timeout = verify_timeout
+        self._state = InstallState.from_installation(installation)
+        self._this_file = Path(__file__)
+
+    @classmethod
+    # pragma: no cover
+    def current(cls, ws: WorkspaceClient):
+        installation = Installation.current(ws, PRODUCT_INFO.product_name())
+        config = installation.load(MorphConfig)
+        prompts = Prompts()
+        timeout = timedelta(minutes=2)
+        return WorkspaceInstallation(config, installation, ws, prompts, timeout)
+
+    def run(self):
+        logger.info(f"Installing Remorph v{PRODUCT_INFO.version()}")
+        self._installation.save(self._config)
+        logger.info("Installation completed successfully! Please refer to the  for the next steps.")
+
+    def uninstall(self):
+        if self._prompts and not self._prompts.confirm(
+            "Do you want to uninstall remorph from the workspace too, this would remove remorph project folder"
+        ):
+            return
+        # TODO: this is incorrect, fetch the remote version (that appeared only in Feb 2024)
+        logger.info(f"Deleting Remorph v{PRODUCT_INFO.version()} from {self._ws.config.host}")
+        try:
+            self._installation.files()
+        except NotFound:
+            logger.error(f"Check if {self._installation.install_folder()} is present")
+            return
+        self._installation.remove()
+        logger.info("UnInstalling Remorph complete")
+
+
+class CatalogSetup:
+    def __init__(self, ws: WorkspaceClient):
+        self._ws = ws
+
+    def create(self, name: str):
+        logger.debug(f"Creating Catalog {name}")
+        catalog_info = self._ws.catalogs.create(name, comment="Created as part of Remorph setup")
+        logger.info(f"Catalog {name} created")
+        return catalog_info
+
+    def get(self, name: str):
+        try:
+            logger.debug(f"Searching for Catalog {name}")
+            catalog_info = self._ws.catalogs.get(name)
+            logger.info(f"Catalog {name} found")
+            return catalog_info.name
+        except NotFound as err:
```

Comment:
don't you want this exception to bubble up?

Reply:
Updated the code